utils/create_dummy_models.py

```diff
@@ -420,24 +420,52 @@ def get_tiny_config(config_class, model_class=None, **model_tester_kwargs):
         error = f"Tiny config not created for {model_type} - no model tester is found in the testing module."
         raise ValueError(error)
 
+    # CLIP-like models have `text_model_tester` and `vision_model_tester`, and we need to pass `vocab_size` to
+    # `text_model_tester` via `text_kwargs`. The same trick is also necessary for `Flava`.
+
+    if "vocab_size" in model_tester_kwargs:
+        if "text_kwargs" in inspect.signature(model_tester_class.__init__).parameters.keys():
+            vocab_size = model_tester_kwargs.pop("vocab_size")
+            model_tester_kwargs["text_kwargs"] = {"vocab_size": vocab_size}
+
     # `parent` is an instance of `unittest.TestCase`, but we don't need it here.
     model_tester = model_tester_class(parent=None, **model_tester_kwargs)
 
     if hasattr(model_tester, "get_pipeline_config"):
-        return model_tester.get_pipeline_config()
+        config = model_tester.get_pipeline_config()
     elif hasattr(model_tester, "prepare_config_and_inputs"):
         # `PoolFormer` has no `get_config` defined. Furthermore, it's better to use `prepare_config_and_inputs` even if
         # `get_config` is defined, since there might be some extra changes in `prepare_config_and_inputs`.
-        return model_tester.prepare_config_and_inputs()[0]
+        config = model_tester.prepare_config_and_inputs()[0]
     elif hasattr(model_tester, "get_config"):
-        return model_tester.get_config()
+        config = model_tester.get_config()
     else:
         error = (
             f"Tiny config not created for {model_type} - the model tester {model_tester_class.__name__} lacks"
             " necessary method to create config."
         )
         raise ValueError(error)
 
+    # make sure this is long enough (some model tester has `20` for this attr.) to pass `text-generation`
+    # pipeline tests.
+    max_positions = []
+    for key in ["max_position_embeddings", "max_source_positions", "max_target_positions"]:
+        if getattr(config, key, 0) > 0:
+            max_positions.append(getattr(config, key))
+        if getattr(config, "text_config", None) is not None:
+            if getattr(config.text_config, key, None) is not None:
+                max_positions.append(getattr(config.text_config, key))
+    if len(max_positions) > 0:
+        max_position = max(200, min(max_positions))
+        for key in ["max_position_embeddings", "max_source_positions", "max_target_positions"]:
+            if getattr(config, key, 0) > 0:
+                setattr(config, key, max_position)
+            if getattr(config, "text_config", None) is not None:
+                if getattr(config.text_config, key, None) is not None:
+                    setattr(config.text_config, key, max_position)
+
+    return config
```

Comment:
This is to avoid the failing in `TextGenerationPipelineTests` at
```python
outputs = text_generator("This is a test" * 500, handle_long_generation="hole", max_new_tokens=20)
```
(failing happens if a model tester has `max_position_embeddings=20`)

Also taking `max_source_positions` and `max_target_positions` into account (e.g. whisper)